packages/cli-kit/src/log.test.ts

```diff
@@ -0,0 +1,156 @@
+import {file} from './index.js'
+import {isUnitTest} from './environment/local.js'
+import {initiateLogging, LinesTruncatorTransformer} from './log.js'
+import {generateRandomUUID} from './id.js'
+import {join} from './path.js'
+import fs from 'fs-extra'
+import {beforeEach, describe, expect, it, vi} from 'vitest'
+import {appendFileSync} from 'fs'
+import {EOL} from 'os'
+import {randomBytes} from 'crypto'
+
+const KB_256 = 0.256 * 1024 * 1024
+const MB_2 = 2 * 1024 * 1024
+const MB_7 = 7 * 1024 * 1024
+
+beforeEach(() => {
+  vi.mock('./id')
+  vi.mock('./environment/local', async () => {
+    return {
+      isUnitTest: vi.fn(),
+      isVerbose: vi.fn(() => false),
+    }
+  })
+})
+
+describe.skip('initiateLogging', () => {
+  it('when unit testing should not initiate log file', async () => {
+    // Given
+    vi.mocked(isUnitTest).mockReturnValue(true)
+
+    // When
+    await initiateLogging()
+
+    // Then
+    expect(generateRandomUUID).not.toBeCalled()
+  })
+
+  it('when log file is not big enough no truncation is executed', async () => {
+    await file.inTemporaryDirectory(async (tmpDir) => {
+      // Given
+      vi.mocked(isUnitTest).mockReturnValue(false)
+      vi.mocked(generateRandomUUID).mockReturnValue('random-uuid')
+      const logPath = join(tmpDir, 'shopify.cli.log')
+      createLogFile(logPath, KB_256, MB_2)
+      const fileSize = fileSizeSync(logPath)
+
+      // When
+      await initiateLogging({logDir: tmpDir, override: true})
+
+      // Then
+      expect(fileSize).toEqual(fileSizeSync(logPath))
+    })
+  })
+
+  it('when log file is big enough it should be truncated', async () => {
+    await file.inTemporaryDirectory(async (tmpDir) => {
+      // Given
+      vi.mocked(isUnitTest).mockReturnValue(false)
+      vi.mocked(generateRandomUUID).mockReturnValue('random-uuid')
+      const logPath = join(tmpDir, 'shopify.cli.log')
+      createLogFile(logPath, KB_256, MB_7)
+      const fileSize = fileSizeSync(logPath)
+
+      // When
+      await initiateLogging({logDir: tmpDir, override: true})
+
+      // Then
+      expect(fileSize).toBeGreaterThan(fileSizeSync(logPath))
```

Comment:
Could you add an expect to check that `fileSizeSync(logPath)` is less than the maximum allowed size?